src/order_cmd.cpp

```diff
@@ -1786,6 +1787,11 @@ void CheckOrders(const Vehicle *v)
 		if (v->orders.list != nullptr) v->orders.list->DebugCheckSanity();
 #endif
 
+		/* If the vehicle is not an aircraft and breakdowns are enabled, check if it has a depot order. If not, check if we should send a warning. */
+		if (message == INVALID_STRING_ID && !v->HasDepotOrder() && v->type != VEH_AIRCRAFT && _settings_client.gui.no_depot_order_warn && _settings_game.difficulty.vehicle_breakdowns != 0) {
```

Comment:
I would move the `!v->HasDepotOrder()` to the end of the condition expression, so it is only going to loop through the orders looking for a depot order when breakdowns and this warning are enabled.